python/tank/descriptor/descriptor_installed_config.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from __future__ import with_statement
+
+import os
+
+from .descriptor_config import ConfigDescriptor
+from . import constants
+from .. import pipelineconfig_utils
+from .. import LogManager
+from ..util import ShotgunPath
+
+from ..errors import TankNotPipelineConfigurationError, TankFileDoesNotExistError, TankInvalidCoreLocationError
+
+log = LogManager.get_logger(__name__)
+
+
+class InstalledConfigDescriptor(ConfigDescriptor):
+    """
+    Descriptor that describes an installed Toolkit Configuration. An installed configuration
+    is what we otherwise refer to as a classic pipeline configuration, which is a pipeline
+    configuration is that installed in a folder on the network, which contains a copy of the
+    environment files, a copy of core and all the bundles required by that pipeline configuration.
+    It supports localized as well as shared core and as such, the interpreter files can be found
+    inside the configuration folder or alongside the shared core.
+    """
+
+    def __init__(self, io_descriptor):
+        super(InstalledConfigDescriptor, self).__init__(io_descriptor)
+        self._io_descriptor.set_missing_manifest_supported(True)
+        self._io_descriptor.set_is_copiable(False)
+
+        # As it is possible to create an installed config descriptor that is not actually on disk,
+        # get_path might return None. Only attempt to set the manifest location when the config
+        # actually exists. Not setting it when it doesn't exist isn't a big deal as it won't be
+        # use anyway.
+        path = self.get_path()
+        if path:
+            self._io_descriptor.set_manifest_location_override(
```

Comment:
Not convinced. The last thing I want is for io_descriptor classes to become a mess of code that needs to test for various layouts inside the bundle. I've considered passing the manifest location as a value to the `__init__` method instead and that would have generated even more noise in the code.